### ISSUE TITLE
Offcanvas and sticky top conflict

### ISSUE THREAD

**Issue by kritsky** [twbs/bootstrap#37920]
### Prerequisites

- [x] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [x] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [x] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

When I click on menu button that shows offcanvas menu, another div doesn't fade out as everything else on the page.

I've made an example with base code from getbootstrap:
https://codepen.io/mess2525/pen/vYapjyr

### Reduced test cases

When I am not using sticky-top its ok

### What operating system(s) are you seeing the problem on?

macOS

### What browser(s) are you seeing the problem on?

Safari

### What version of Bootstrap are you using?

5.3.0

**Comment by louismaximepiton** [comment]
Hey there, 

I think the main issue is the stacking context used here. 

Indeed, as [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context) mentions, the issue here is that the 2 `.sticky-top` used are seen on the same stacking context since there isn't any new stacking context between. 

For instance, the nav and the box are seen on the same layer with the same z-index. The markup order takes the advantages and the box layer is shown over the nav layer because the box comes after the nav in the markup. As an example here, if there was a `.position-relative` or `.position-sticky` on `div.container` it would start a new layer. There wouldn't be any issue since the nav layer would have a z-index of 1020 and the container layer would have a z-index of auto.

I don't know if this was clear enough but maybe take a look at MDN to better understand.

tl;dr: use `.position-relative` on `div.container` correct this issue.

IMO, there isn't anything to do on Bootstrap side.

**Comment by kritsky** [comment]
I've added `.position-relative` to <div class="container position-relative">. It doesn't help. Updated codepen

**Comment by louismaximepiton** [comment]
Ah yes you're right, sorry for that. I forgot to add my whole thought: it depends on position value, you may need to specify `z-index`. Here are some of the tricks (found on [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context)) you could dig:

> Element with a [position](https://developer.mozilla.org/en-US/docs/Web/CSS/position) value absolute or relative and [z-index](https://developer.mozilla.org/en-US/docs/Web/CSS/z-index) value other than auto.

Add a `z-index` on the actual implementation.

> Element with a [position](https://developer.mozilla.org/en-US/docs/Web/CSS/position) value fixed or sticky (sticky for all mobile browsers, but not older desktop browsers).

Change the actual `.position-relative` to `.position-sticky/fixed`. (might be bad depending on your implementation)

> Element with an [isolation](https://developer.mozilla.org/en-US/docs/Web/CSS/isolation) value isolate.

Add `style="isolation: isolate;` to your container since Bootstrap don't provide such utility. (you may remove `.position-relative`)

> Element with a [contain](https://developer.mozilla.org/en-US/docs/Web/CSS/contain) value of layout, or paint, or a composite value that includes either of them (i.e. contain: strict, contain: content).

Add `style="contain: layout;"` to your container since Bootstrap don't provide such utility. (you may remove `.position-relative`)

**Comment by kritsky** [comment]
Is it possible for you to show a working example?

**Comment by louismaximepiton** [comment]
Here it is https://codepen.io/louismaximepiton/pen/OJweyzV

**Comment by kritsky** [comment]
Great, it works! Thank you

**Comment by kritsky** [comment]
Find another bug, now with modal. z-0 at container conflicts with it
Check by clicking: https://codepen.io/mess2525/pen/vYapjyr

**Comment by louismaximepiton** [comment]
You might need to do the trick to the nearest element of the `.sticky-top` here it is: https://codepen.io/louismaximepiton/pen/XWPedNj
